pinot-segment-local/src/main/java/org/apache/pinot/segment/local/function/InbuiltFunctionEvaluator.java

```diff
@@ -68,17 +67,23 @@ private ExecutableNode planExecution(ExpressionContext expression) {
           childNodes[i] = planExecution(arguments.get(i));
         }
         String functionName = function.getFunctionName();
-        FunctionInfo functionInfo = FunctionRegistry.getFunctionInfo(functionName, numArguments);
-        if (functionInfo == null) {
-          if (FunctionRegistry.containsFunction(functionName)) {
-            throw new IllegalStateException(
-              String.format("Unsupported function: %s with %d parameters", functionName, numArguments));
-          } else {
-            throw new IllegalStateException(
-              String.format("Unsupported function: %s not found", functionName));
-          }
+        switch (functionName) {
+          case "and":
+          case "or":
+            return new ConjugationExecutionNode(functionName, childNodes);
```

Comment:
Suggest adding 3 execution nodes: `AndExecutionNode`, `OrExecutionNode`, `NotExecutionNode`. IMO no need to add the concept of `ConjugateFunction`

Reply:
Ack

Reply:
Ack